### ISSUE TITLE
Google approval_prompt enforcement is breaking login

### ISSUE THREAD

**Issue by jamesgraham** [thephpleague/oauth2-google#87]
It looks like in the past couple days, Google are now blocking requests with an approval_prompt sent e.g. `approval_prompt=auto`.

The base oauth2-client decides to set this as a default. This is not good for us.
Unless you set `prompt` to a value, your login will now break.

I suggest forcing approval_prompt to null in all cases here:
https://github.com/thephpleague/oauth2-google/blob/cfe9b0ae69eb5dbac64a282f04410b731f6b9c04/src/Provider/Google.php#L70

Drupal team also found this out: https://www.drupal.org/project/social_auth_google/issues/3129494
Cannot see any notices from Google regarding this.

**Comment by majkel89** [comment]
;-) same for me. Maybe it is problem with Google itself ?

I' monkey patching mine code for now

```
$redirectUrl = str_replace('approval_prompt=auto&', '', $redirectUrl);
```

**Comment by hugovk** [comment]
See also https://github.com/thephpleague/oauth2-client/issues/831, and thephpleague/oauth2-client is the library that Drupal module uses, but would be good to fix upstream too.

**Comment by sasa1007** [comment]
So is there some solution for overriding - I am using Symfony

**Comment by jamesgraham** [comment]
> So is there some solution for overriding - I am using Symfony

Try to set `prompt` to an allowed value e.g. consent

**Comment by sasa1007** [comment]
> > So is there some solution for overriding - I am using Symfony
> 
> Try to set `prompt` to an allowed value e.g. `consent` or set `approval_prompt` to null so it wipes out the base client trying to set it to `auto`

It was null but not working

**Comment by Chrisissorry** [comment]
Using the library in conjunction with https://github.com/knpuniversity/oauth2-client-bundle, both `prompt` and `approval_prompt` are not valid configuration params. Using Symfony 4.4.2 here.

**Comment by sasa1007** [comment]
I found solution in

namespace League\OAuth2\Client\Provider;

line 345
       $options += [
            'response_type'   => 'code',
            'prompt' => 'consent'
        ];

**Comment by majkel89** [comment]
@sasa1007  I fixed the problem with

```
$client->redirect([], [
  'prompt' => 'consent',
])
```

**Comment by Chrisissorry** [comment]
Yes, same works for us: 

```    
public function connectAction(): RedirectResponse
    {
        return $this->clientRegistry
            ->getClient('google')
            ->redirect(self::$scopes, ['prompt' => 'consent']);
    }
```

**Comment by sasa1007** [comment]
@majkel89  working thanks

**Comment by shadowhand** [comment]
The `approval_prompt=auto` option was removed in #66. Is this still an issue?

**Comment by shadowhand** [comment]
Never mind, I see the problem now. The default `approval_prompt` setting is part of the default client parameters.

**Comment by shadowhand** [comment]
Fixed in version 3.0.3.
